source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -945,6 +962,38 @@ void InjaTransformer::transform(Http::RequestOrResponseHeaderMap &header_map,
   } else if (merged_extractors_to_body_) {
     std::string output = json_body.dump();
     maybe_body.emplace(output);
+  } else if (!merge_templates_.empty()) {
+
+    for (const auto &merge_template : merge_templates_) {
+      const std::string &name = std::get<0>(merge_template);
+      
+      // prepare variables for non-advanced_templates_ scenario
+      absl::string_view name_to_split;
+      json* current = nullptr;
+      // if (!advanced_templates_) {
+      name_to_split = name;
+      current = &json_body;
+      for (size_t pos = name_to_split.find("."); pos != std::string::npos;
+          pos = name_to_split.find(".")) {
+        auto &&field_name = name_to_split.substr(0, pos);
+        current = &(*current)[std::string(field_name)];
+        name_to_split = name_to_split.substr(pos + 1);
+      }
+      const auto rendered = instance_->render(std::get<2>(merge_template));
+      // Do not overwrite with empty unless specified
+      if (rendered.size() > 0 || std::get<1>(merge_template)) {
+        try {
+          auto rendered_json = json::parse(rendered);
+          (*current)[std::string(name_to_split)] = rendered_json;
+        } catch (const std::exception &e) {
+          ENVOY_STREAM_LOG(debug, "failed to parse merge_json_key output: {}", callbacks, e.what());
+          (*current)[std::string(name_to_split)] = rendered;
```

Comment:
I'm not a fan of during different logic in a way that's not predictable (i.e. if we had an error parsing the json). can this be explicit config? i.e. parse template as json or not?